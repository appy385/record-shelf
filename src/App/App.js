import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import Sync from '../pages/Sync/Sync';
import './App.scss';

const App = () => (
  <BrowserRouter>
    <div className="app-container">
      <Switch>
        <Route path="/"><Sync /></Route>
      </Switch>
    </div>
  </BrowserRouter>
);

export default App;
