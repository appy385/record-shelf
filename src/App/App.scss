.app-container {
    display: flex;
    flex-direction: column;
    margin: 0.5rem;
    border: 1px solid black;
}