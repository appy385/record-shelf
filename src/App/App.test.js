import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import App from './App';

describe(App.name, () => {
  test('shoudl match snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });
  test('should render button sync', () => {
    render(<App />);
    const syncButton = screen.getByText('Sync');
    expect(syncButton.tagName).toBe('BUTTON');
  });
  test('should go to /songs route on sync button click ', () => {
    render(<App />);
    const syncButton = screen.getByText('Sync');
    fireEvent.click(syncButton);
    expect(document.location.href).toBe('http://localhost/songs');
  });
});
