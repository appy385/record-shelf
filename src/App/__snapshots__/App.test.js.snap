// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` shoudl match snapshot 1`] = `
<div>
  <div
    class="app-container"
  >
    <div
      class="sync-container"
    >
      <span>
        :((
      </span>
      <span>
        Seems a bit empty in here...
      </span>
      <button
        class="sync-button"
        type="button"
      >
        Sync
      </button>
    </div>
  </div>
</div>
`;
